src/test.rs

```diff
@@ -1,25 +1,38 @@
 use kuchiki::traits::*;
-use reqwest;
-use serde_json::{self, Value};
+use kuchiki::NodeRef;
 use std::fs;
 use std::path::Path;
 
 use crate::metadata::metadata::Metadata;
 use crate::spec::Spec;
 
-pub fn clean_html(code: String) -> Result<String, Box<dyn std::error::Error>> {
-    let client = reqwest::blocking::Client::new();
+// Compare DOM trees recursively.
+fn is_equal(lhs: &NodeRef, rhs: &NodeRef) -> bool {
+    if lhs.data() != rhs.data() {
+        return false;
+    }
 
-    let res = client
-        .post("https://www.10bestdesign.com/dirtymarkup/api/html")
-        .form(&[("code", code), ("indent", "2".to_owned())])
-        .send()?
-        .text()?;
+    // remove text nodes
+    let lhs_children = lhs
+        .children()
+        .filter(|lc| lc.as_text().is_none())
+        .collect::<Vec<NodeRef>>();
+    let rhs_children = rhs
+        .children()
+        .filter(|rc| rc.as_text().is_none())
+        .collect::<Vec<NodeRef>>();
+
+    if lhs_children.len() != rhs_children.len() {
+        return false;
+    }
 
-    let val: Value = serde_json::from_str(&res)?;
-    let cleaned = val["clean"].as_str().unwrap();
+    for (lc, rc) in lhs_children.iter().zip(rhs_children.iter()) {
```

Comment:
You could write this like `lhs_children.iter().zip(rhs_children.iter()).all(is_equal)`, I think.

Reply:
It seems like the `all` would only return `bool`. Is that possible to get `Result<(), CompareError>` with that?